src/Operators/arrayinterpolate.jl

```diff
@@ -0,0 +1,165 @@
+    Nz = Spaces.nlevels(space)
+
+    velem = Meshes.containing_element(vert_mesh, zcoord)
+    ξ3, = Meshes.reference_coordinates(vert_mesh, velem, zcoord)
+    if space isa Spaces.FaceExtrudedFiniteDifferenceSpace
+        v_lo = velem - half
+        v_hi = velem + half
+    elseif space isa Spaces.CenterExtrudedFiniteDifferenceSpace
+        if ξ3 < 0
+            if Topologies.isperiodic(Spaces.vertical_topology(space))
+                v_lo = mod1(velem - 1, Nz)
+            else
+                v_lo = max(velem - 1, 1)
+            end
+            v_hi = velem
+            ξ3 = ξ3 + 1
+        else
+            v_lo = velem
+            if Topologies.isperiodic(Spaces.vertical_topology(space))
+                v_hi = mod1(velem + 1, Nz)
+            else
+                v_hi = min(velem + 1, Nz)
+            end
+            ξ3 = ξ3 - 1
+        end
+    end
+    f_lo = interpolate_slab(field, Fields.SlabIndex(v_lo, h), Is)
+    f_hi = interpolate_slab(field, Fields.SlabIndex(v_hi, h), Is)
+    return ((1 - ξ3) * f_lo + (1 + ξ3) * f_hi) / 2
+end
+
+"""
+    interpolate_array(field, xpts, ypts)
+    interpolate_array(field, xpts, ypts, zpts)
+
+Interpolate a field to a regular array using pointwise interpolation.
+
+This is primarily used for plotting and diagnostics.
+
+# Examples
+
+```julia
+longpts = range(Geometry.LongPoint(-180.0), Geometry.LongPoint(180.0), length = 21)
+latpts = range(Geometry.LatPoint(-80.0), Geometry.LatPoint(80.0), length = 21)
+zpts = range(Geometry.ZPoint(0.0), Geometry.ZPoint(1000.0), length = 21)
+
+interpolate_array(field, longpts, latpts, zpts)
+```
+
+!!! note
+    Hypsography is not currently handled correctly.
+"""
+function interpolate_array end
+
+function interpolate_array(
+    field::Fields.ExtrudedFiniteDifferenceField,
+    xpts,
+    zpts,
+)
+    space = axes(field)
+    @assert ClimaComms.context(space) isa ClimaComms.SingletonCommsContext
+
+    horz_topology = Spaces.topology(space)
+    horz_mesh = horz_topology.mesh
+
+    T = eltype(field)
+    array = zeros(T, length(xpts), length(zpts))
+
+    FT = Spaces.undertype(space)
+    for (ix, xcoord) in enumerate(xpts)
+
+        hcoord = xcoord
+        helem = Meshes.containing_element(horz_mesh, hcoord)
+        ξ1, = Meshes.reference_coordinates(horz_mesh, helem, hcoord)
+        quad = Spaces.quadrature_style(space)
+        quad_points, _ = Spaces.Quadratures.quadrature_points(FT, quad)
+        WI1 = Spaces.Quadratures.interpolation_matrix(SVector(ξ1), quad_points)
+        h = helem
+
+        for (iz, zcoord) in enumerate(zpts)
+            array[ix, iz] = interpolate_slab_level(field, h, (WI1,), zcoord)
+        end
+    end
+    return array
+end
+
+function interpolate_array(
+    field::Fields.ExtrudedFiniteDifferenceField,
+    xpts,
+    ypts,
+    zpts,
+)
+    space = axes(field)
+    @assert ClimaComms.context(space) isa ClimaComms.SingletonCommsContext
+
+    horz_topology = Spaces.topology(space)
```

Comment:
```suggestion
    horz_topology = Spaces.topology(Spaces.horizontal_space(space))
```
Don't we need to extract the horizontal space for this?

Reply:
(same for above)